Document changes related to continuous localization (#86)
freedomofpress/securedrop-dev-docs: docs/i18n.rst

Reviewer: This is looking good.  The section "Push your branch and create a pull request" can also go!

directly. Translation happens on our `Weblate`_ server, which is
configured to use a fork of the `main SecureDrop repository`_.

As string changes are merged into the ``develop`` branch
in the `main SecureDrop repository`_, the changes will automatically

> appear in Weblate, and translation can begin. Translations take
> place continuously, at any time. Translations for a release are
> reviewed and finalized during the release process.

Reviewer: Some things to clarify and/or expand:

```suggestion
appear in Weblate, and translation can begin. Translation and review can take
place continuously, at any time. Translations for supported languages [link] are
reviewed and finalized during the release process.
```

As in <https://github.com/freedomofpress/securedrop-client/wiki/Internationalization>, I think our goal should be that translations get code-merged more rather than less often.  This will mean (a) [some translations get code-merged before they've been translation-reviewed](https://github.com/freedomofpress/securedrop-client/pull/1548#discussion_r1156375077) and (b) some translations will be merged for unsupported languages, before language support is evaluated as part of the release process.

> files should also be updated by running ``make translate`` in the root

Reviewer: Per below, we'll name this target more explicitly going forward (and elsewhere in this document):

```suggestion
files should also be updated by running ``make extract-strings`` in the root
```

Yet to be determined: whether this is enforced at commit-time or only in CI.

of the SecureDrop working copy.

> The ``translate`` target gathers source strings, then updates the 
> `.pot`_ files for the SecureDrop server code and the desktop icons, as well as
> the `.po`_ files for each language.

Reviewer: ```suggestion
The ``extract-strings`` target gathers source strings, then updates the 
`.pot`_ files for the SecureDrop server code and the desktop icons.
```

The new approach defers the `.pot`-to-`.po` merge to Weblate.  See freedomofpress/securedrop-client#1657 for a tidy example.  This would also remove the paragraph beginning "Then review...".

Author: That makes sense, thanks for clarifying!

``l10n: update strings to be translated for vX.Y.Z`` pull request as a

> template.

Reviewer: This paragraph can be removed: source strings will be updated in the same branch (if not the very same commits) in which the enclosing source code is changed.

.. _merge_develop_to_weblate:

> Merge develop into the Weblate fork

Reviewer: A number of steps here can disappear entirely:
* Merge develop into the Weblate fork (and references in "Two weeks before the release: string freeze")
* Merge translations back to develop (and references in "Release day")
* Merge translations back to develop

    consider pairing on this ceremony, both for knowledge-sharing and so that
    the intermediate pull requests can be reviewed and merged promptly.

When features for a new SecureDrop release are frozen, so are the source

> strings. The localization manager should apply any source-string suggestions

Author: Should `source-string` be hyphenated here?

Reviewer: I believe so: "source-string" modifies "suggestions", rather than "source" modifying "string suggestions".

.. code:: sh

>     $ securedrop/bin/dev-shell make extract-strings

Reviewer: This section describes the `.po`→`.mo` compilation rather than the `.py`→`.po[t]` extraction.  But it can go away too: if we're not letting Weblate compile our `.mo` files, it'll be part of `make dev` and/or `dev-deps` (TBD).
